Store/BLL/Services/ProductService.cs

```diff
@@ -29,24 +31,29 @@ public ProductModel Update(ProductModel productModel)
                 throw new ArgumentException("Product has incorrect data", nameof(productModel));
             }
 
-            if (_repository.GetById(productModel.Id) == null)
+            if (_productRepository.GetById(productModel.Id) == null)
             {
                 throw new ArgumentException("Product does not exist", nameof(productModel));
             }
 
-            Product productTemp = _repository.GetByName(productModel.Name);
+            if (_categoryRepository.Get(productModel.CategoryId) == null)
```

Comment:
IMHO: a lot of IFs for validate. maybe u can try to make some kind of universal helper for validation. maybe you can look through Generics